core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java

```diff
@@ -389,9 +398,16 @@ private void initAfterDeserialisation(MuleContext muleContext) throws MuleExcept
         }
 
       }
-      // Can be null if service call originates from MuleClient
-      if (context.getOriginatingFlowName() != null) {
```

Comment:
From my POV, this was more about understanding the entry point (into mule) of an event, for notifications etc., so i'd say yes.  Unless there is a connector source reference that replaces this as it also identifies flow